### PR TITLE
MINOR:Refactor the metric names into constants in ReplicaManager

### PR DESCRIPTION
**Pull request by hudeqi**
Inspired by #13623, the code of the metric in ReplicaManager is optimized (to improve the commit of #13471).

### CODE REVIEW

**Comment by hudeqi** [comment]
and this, thanks! @divijvaidya @dajac

**Review by kirktrue — COMMENTED** [review]
Mostly reviewing this to learn more about the broker side of things. Seems like a straightforward clean up.

**Review by machi1990 — APPROVED** [review]
lgtm

File: core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala
Comment (divijvaidya) [inline core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala]: echo'ing a comment that David provided in https://github.com/apache/kafka/pull/13623#discussion_r1182595446, we can replace the `anyString()` with actual names to verify that the argument passed are correct.

File: core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala
Comment (divijvaidya) [inline core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala]: This line could be removed. The line below is verifying that all entries in `GaugeMetricNames` are having a function invocation.

File: core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala
Comment (divijvaidya) [inline core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala]: ditto

File: core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala
Comment (divijvaidya) [inline core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala]: ditto

**Review by divijvaidya — CHANGES_REQUESTED** [review]

**Comment by divijvaidya** [comment]
@hudeqi you requested the review today but I think there are some existing comments above that have not yet been addressed

**Comment by hudeqi** [comment]
Hi, @divijvaidya,  it has been updated according to your comment, thank you!

**Review by divijvaidya — COMMENTED** [review]
Looks good to me. One minor thing is to update the PR name (since we use the same for commit messages). As I understand, we are not really optimizing anything here. We are refactoring the metric names into constants. 

Can you please change the PR description to reflect the same?

**Comment by hudeqi** [comment]
> Looks good to me. One minor thing is to update the PR name (since we use the same for commit messages). As I understand, we are not really optimizing anything here. We are refactoring the metric names into constants. 
> 
> Can you please change the PR description to reflect the same? 

ok, I see, updated the pr name. @divijvaidya

**Comment by divijvaidya** [comment]
I triggered the CI tests again since last run wasn't complete. Will merge this in after they are successful.

**Comment by hudeqi** [comment]
> I triggered the CI tests again since last run wasn't complete. Will merge this in after they are successful.

It seems that the failure of CI check has nothing to do with this pr. @divijvaidya

**Comment by divijvaidya** [comment]
> It seems that the failure of CI check has nothing to do with this pr. @divijvaidya

Hmm...not sure about that. There are 99 tests failing and I am not comfortable saying that they are not because of this PR. Let's wait a day or 2 for CI to stabilize, after that we can merge from trunk again and try the CI for this PR.

**Comment by hudeqi** [comment]
> > It seems that the failure of CI check has nothing to do with this pr. @divijvaidya
> 
> Hmm...not sure about that. There are 99 tests failing and I am not comfortable saying that they are not because of this PR. Let's wait a day or 2 for CI to stabilize, after that we can merge from trunk again and try the CI for this PR.

Is this CI test often unstable? I've hardly ever seen anything that can be completely checked successfully. 😂

**Comment by divijvaidya** [comment]
Failing tests are unrelated. These are frequently failing tests in trunk.
```
[Build / JDK 17 and Scala 2.13 / org.apache.kafka.connect.mirror.integration.MirrorConnectorsIntegrationBaseTest.testReplicateSourceDefault()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-13705/11/testReport/junit/org.apache.kafka.connect.mirror.integration/MirrorConnectorsIntegrationBaseTest/Build___JDK_17_and_Scala_2_13___testReplicateSourceDefault__/)
[Build / JDK 17 and Scala 2.13 / kafka.api.ConsumerBounceTest.testClose()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-13705/11/testReport/junit/kafka.api/ConsumerBounceTest/Build___JDK_17_and_Scala_2_13___testClose__/)
[Build / JDK 17 and Scala 2.13 / kafka.api.TransactionsTest.testBumpTransactionalEpoch(String).quorum=kraft](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-13705/11/testReport/junit/kafka.api/TransactionsTest/Build___JDK_17_and_Scala_2_13___testBumpTransactionalEpoch_String__quorum_kraft/)
[Build / JDK 17 and Scala 2.13 / kafka.zk.ZkMigrationIntegrationTest.[1] Type=ZK, Name=testNewAndChangedTopicsInDualWrite, MetadataVersion=3.4-IV0, Security=PLAINTEXT](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-13705/11/testReport/junit/kafka.zk/ZkMigrationIntegrationTest/Build___JDK_17_and_Scala_2_13____1__Type_ZK__Name_testNewAndChangedTopicsInDualWrite__MetadataVersion_3_4_IV0__Security_PLAINTEXT/)
[Build / JDK 11 and Scala 2.13 / org.apache.kafka.connect.mirror.integration.IdentityReplicationIntegrationTest.testReplicateSourceDefault()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-13705/11/testReport/junit/org.apache.kafka.connect.mirror.integration/IdentityReplicationIntegrationTest/Build___JDK_11_and_Scala_2_13___testReplicateSourceDefault__/)
[Build / JDK 11 and Scala 2.13 / kafka.zk.ZkMigrationIntegrationTest.[1] Type=ZK, Name=testNewAndChangedTopicsInDualWrite, MetadataVersion=3.4-IV0, Security=PLAINTEXT](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-13705/11/testReport/junit/kafka.zk/ZkMigrationIntegrationTest/Build___JDK_11_and_Scala_2_13____1__Type_ZK__Name_testNewAndChangedTopicsInDualWrite__MetadataVersion_3_4_IV0__Security_PLAINTEXT/)
[Build / JDK 11 and Scala 2.13 / org.apache.kafka.controller.QuorumControllerTest.testBalancePartitionLeaders()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-13705/11/testReport/junit/org.apache.kafka.controller/QuorumControllerTest/Build___JDK_11_and_Scala_2_13___testBalancePartitionLeaders__/)
[Build / JDK 8 and Scala 2.12 / kafka.api.ConsumerBounceTest.testClose()](https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-13705/11/testReport/junit/kafka.api/ConsumerBounceTest/Build___JDK_8_and_Scala_2_12___testClose__/)
```

**Pull request merged**
